packages/docgen/scripts/dump-ast.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+const ts = require( 'typescript' );
+const { readFileSync } = require( 'fs' );
+const { inspect } = require( 'util' );
+
+const filename = process.argv[ 2 ];
+
+const sourceFile = ts.createSourceFile(
+	filename,
+	readFileSync( filename ).toString(),
+	ts.ScriptTarget.ESNext
+);
+
+// eslint-disable-next-line no-console
+console.log( inspect( sourceFile, false, null ) );
```

Comment:
Two minor points:

- We should want to document our inline disablings (related: #16795)
- We could probably avoid it if we used `process.stdout.write` instead, which also might be the more "correct" way to arbitrarily write to stdout (?)

Reply:
I changed it to `process.stdout.write`.